### DISCUSSION TITLE
Octobot <> NinjaTrader

### DISCUSSION THREAD

**Discussion by deadbrainviv** [Drakkar-Software/OctoBot#2024] [Ideas]
Hello Team,

First of all thanks for writing such an amazing tool as an opensource project. I have been using octobot for my organization's research and development tasks. 

I have a question regarding how to proceed with an integration with NinjaTrader and Octobot for automating the trading through octobot. 

Any advise or suggestions would be a starting point for me to write the API and if there is any snippets in place that would be wonderful 

Regards
Vivek

**Comment by Herklos** [comment]
Hi @deadbrainviv!
I think the best thing you can do is to create a tentacle to integrate NinjaTrader api into OctoBot.
You should have a look at https://octobot.click/gh-developer where tentacle development is explained.

**Reply by deadbrainviv** [reply]
Thanks for your suggestions on building a custom tentacle. But as ninja trader is also capable for automated trading,  the system is not showing any direct feedback or update being sent from octobot api for creating new trade. Can you suggest any other approach as we are not placing the order directly but just creating an order and triggering the same to ninja trader which would inplace do the actual order placing
